pydropsonde/circles.py

```diff
@@ -16,5 +21,138 @@ class Circle:
     segment_id: str
 
     def dummy_circle_function(self):
-        print(self.flight_id, self.segment_id)
+        print(self.flight_id, self.segment_id, self.platform_id)
+        return self
+
+    def get_xy_coords_for_circles(self):
+        x_coor = (
+            self.circle_ds.lon * 111.320 * np.cos(np.radians(self.circle_ds.lat)) * 1000
+        )
+        y_coor = self.circle_ds.lat * 110.54 * 1000
+
+        # converting from lat, lon to coordinates in metre from (0,0).
+
+        c_xc = np.full(np.size(x_coor, 1), np.nan)
+        c_yc = np.full(np.size(x_coor, 1), np.nan)
+        c_r = np.full(np.size(x_coor, 1), np.nan)
+
+        for j in range(np.size(x_coor, 1)):
+            a = ~np.isnan(x_coor.values[:, j])
+            if a.sum() > 4:
+                c_xc[j], c_yc[j], c_r[j], _ = cf.least_squares_circle(
+                    [
+                        (x, y)
+                        for x, y in zip(x_coor.values[:, j], y_coor.values[:, j])
+                        if ~np.isnan(x)
+                    ]
+                )
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=RuntimeWarning)
+            circle_y = np.nanmean(c_yc) / (110.54 * 1000)
+            circle_x = np.nanmean(c_xc) / (
+                111.320 * np.cos(np.radians(circle_y)) * 1000
+            )
+            circle_diameter = np.nanmean(c_r) * 2
+
+        xc = [None] * len(x_coor.T)
+        yc = [None] * len(y_coor.T)
+
+        xc = np.mean(x_coor, axis=0)
+        yc = np.mean(y_coor, axis=0)
+
+        delta_x = x_coor - xc  # *111*1000 # difference of sonde long from mean long
+        delta_y = y_coor - yc  # *111*1000 # difference of sonde lat from mean lat
+
+        object.__setattr__(self, "circle_lon", circle_x)
```

Comment:
why are you not just adding those to an xarray? 
There is no need to store all the variables as separate attributes to the class. there can just be one xarray to which variables are added. Or is there a reason you need those separate?

Reply:
Yes just related to the other comment, I've been simply following JOANNE